src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -1976,6 +1981,43 @@ public void CanObserveDefaultServicesInServiceCollection()
         Assert.Contains(builder.Services, service => service.ServiceType == typeof(ILogger<>));
     }
 
+    [Fact]
+    public async Task RegisterAuthMiddlewaresCorrectly()
+    {
+        var helloEndpointCalled = false;
+        var helloProtectedEndpointCalled = false;
+
+        var builder = WebApplication.CreateBuilder();
+        builder.WebHost.UseTestServer();
+        builder.Authentication.AddJwtBearer();
+        await using var app = builder.Build();
+
+        app.Use(next =>
+        {
+            return async context =>
+            {
+                // IAuthenticationFeature is added by the authentication middleware
+                // during invocation. This middleware should run after authentication
+                // and be able to access the feature.
+                var authFeature = context.Features.Get<IAuthenticationFeature>();
```

Comment:
- Add a bool, to verify that this code path ran and check if the user is set? 
- I would rewrite this test to use a custom auth handler that makes the user authenticated when a header is set or something that's easy to set with the test server.
- Verify the user is set and the identity's name matches whatever you returned from the auth handler.